### PR TITLE
Inherit from `standardrb` rubocop ruleset

### PR DESCRIPTION
**Pull request by mjankowski**
The core change here is to update the rubocop config to “inherit” from the community-maintained “standardrb” set of rubocop cops, using a technique similar to what was described (linked blog post) in the last config PR - https://github.com/mastodon/mastodon/pull/30407

I had previously want to do this, but standard is (by design!) not configurable when used directly (ie, to fully replace rubocop). However, using this inherit/override approach that changes and we can take it as a starting point instead of going all-or-none. I also think the sheer number of open PRs and forks that would be hit by the churn from a full-scale adoption makes it sort of a non-starter to go all the way anyway.

Goals/benefits:

- Align with general community style/lint standards, outsource bikeshedding, may be beneficial for contributors coming from other projects
- Fewer `rubocop-*` related version bump PRs and corrections needed on our side (the standard gems are released less often, generally wait for rubocop patch releases to settle down and fix minor issues, and generally don’t enable all the new cops, etc)
- Basically "finishes" (one left!) the `_todo` file

Actual changes here:

- Update our initial core config to use `inherit_gem` and pull in rule set from standardrb gems
- Within each of our `.rubocop/*` files, update to remove things which are made not relevant by having standard as base
- Within those same files - add entries where adopting the style from standard would create a comical amount of changes/churn in the project (some rules have thousands of offenses) and we need to override with existing prefs
- Update Gemfile to let the `standard-*` gems pull in their preferred rubocop versions (as of today, I think this actually is latest stable of all of them that we are already on, though this will sometimes lag in future ... but no "downgrades" in this PR)
- Throughout codebase, do once-over on our `rubocop:disable` comments, removing them when no longer relevant. Think I got them all, may need another pass.
- Very small set of actual formatting-only changes which did not get handled in extracted PRs (or maybe are in still-open PRs) … could pull these out if wanted, but not many of them.
- Deleted some of our directory-specific `.rubocop.yml` overrides when the thing they were doing are no longer needed

When it came to deciding what rules to override vs adapt, I sort of used my discretion, but it was also sort of obvious. For things which had hundreds or thousands of offenses, I just added the override to keep our existing style.

There are a few in here which have relatively few offenses that we may want to flip at some point (some of the lowest counts are all in old migrations and would be straightforward-ish to remedy, we have ongoing work to resolve the `Rails/EnvironmentVariableAccess` one via `config_for` PRs, I think `Style/Alias` maybe should flip, etc...), but the bulk of it is probably fine to just stay overridden indefinitely.

This is the last "big" change in this recent series of lint related PRs. The most recent batch were prep for this, extracted from this branch as I tried to get it close enough to open this. I'm hoping this will be the end or close to the end of doing stuff "to" the linter (maybe small followup re: items listed above), as opposed to just having it in place to support other more useful work.

I'll do a pass to link the remaining open PR extracted changes to here after this is open. We could in theory do this in smaller PRs if desired (1 for inherit + base overrides + reformats, another for todo regen, another to remove disables -- or some other approach) or easier to review.

### CODE REVIEW

**Comment by renchap** [comment]
I like the intent, but I am a bit concerned by this: https://github.com/standardrb/standard?tab=readme-ov-file#running-standards-rules-via-rubocop

> Please note that the following usage is not supported and may break at any time. Use at your own risk and please refrain from opening GitHub issues with respect to loading Standard or its plugins' YAML configurations for use by the rubocop CLI.

**Comment by mjankowski** [comment]
Yeah, it's a valid concern ... though I think that warning is there mostly because they don't want to support/debug whatever one-off customizations people might open issues for, and not because it's a particularly large risk.

**Comment by renchap** [comment]
Is there a way for us to switch to `standardrb` entirely at some point, or do you think this can never happen?

**Comment by mjankowski** [comment]
> Is there a way for us to switch to `standardrb` entirely at some point, or do you think this can never happen?

The narrow technical answer is yes we could do it pretty easily, run the autofix, and just run its base config directly from then on. My experience in having teams do this is that everyone hates part of it for like a week or two and then you just get used to it and forget what you cared about and its solid.

The more responsible route if we wanted to go that direction would be to do a pretty thorough pass through as many open PRs as possible ahead of time, merge/close as many as possible, to try to reduce the number of merge conflicts created.

A standard run (with their base config) right now looks like:

<details>
  <summary>Offenses</summary>

```
26618  Style/StringLiterals [Safe Correctable]
4012   Layout/SpaceInsideHashLiteralBraces [Safe Correctable]
2125   Layout/ExtraSpacing [Safe Correctable]
736    Style/TrailingCommaInHashLiteral [Safe Correctable]
447    Style/QuotedSymbols [Safe Correctable]
402    Style/PercentLiteralDelimiters [Safe Correctable]
306    Layout/ArgumentAlignment [Safe Correctable]
149    Style/TrailingCommaInArrayLiteral [Safe Correctable]
121    Style/StringLiteralsInInterpolation [Safe Correctable]
67     Layout/MultilineMethodCallIndentation [Safe Correctable]
67     Style/ArgumentsForwarding [Safe Correctable]
64     Layout/EmptyLineBetweenDefs [Safe Correctable]
52     Layout/ElseAlignment [Safe Correctable]
51     Layout/EndAlignment [Safe Correctable]
47     Layout/IndentationWidth [Safe Correctable]
47     Style/TernaryParentheses [Safe Correctable]
39     Style/EmptyMethod [Safe Correctable]
39     Style/SingleLineMethods [Safe Correctable]
29     Style/Alias [Safe Correctable]
15     Style/RedundantBegin [Safe Correctable]
8      Style/UnlessLogicalOperators
6      Style/GlobalVars
5      Layout/HeredocIndentation [Safe Correctable]
4      Style/LineEndConcatenation [Unsafe Correctable]
3      Layout/SpaceInsideBlockBraces [Safe Correctable]
3      Style/WhenThen [Safe Correctable]
2      Layout/EmptyLineAfterMagicComment [Safe Correctable]
2      Layout/EmptyLinesAroundBlockBody [Safe Correctable]
1      Layout/ClosingHeredocIndentation [Safe Correctable]
1      Layout/EmptyLines [Safe Correctable]
1      Layout/MultilineMethodCallBraceLayout [Safe Correctable]
1      Layout/SpaceBeforeBlockBraces [Safe Correctable]
1      Lint/NonLocalExitFromIterator
1      Lint/RedundantRequireStatement [Unsafe Correctable]
1      Style/ConditionalAssignment [Safe Correctable]
1      Style/SuperWithArgsParentheses [Safe Correctable]
--
35474  Total in 1904 files
```
</details>

My assumption is that due to some combination of a) sheer volume of offenses, b) actual style prefs of the team for some of these items -- that path wasn't desirable. My thinking was that doing what this PR does is a net improvement either way ... if we do someday want to get fully there this will be a good starting point, but even if not this will be a net improvement and simplification of the whole setup.

File: Gemfile.lock [hunk 1]
```diff
@@ -905,7 +919,7 @@ GEM
     zeitwerk (2.6.16)
 
 PLATFORMS
-  ruby
+  arm64-darwin-23
```
Comment (nschonni) [inline Gemfile.lock:922]: Something slipped through during the rebase

**Comment by mjankowski** [comment]
Rebased and updated standard to 1.40.1

With 4.3 out, would love to get feedback on this direction.

**Comment by mjankowski** [comment]
Bumped to standard 1.41.0 which gets the other sub-deps current as well.

**Comment by mjankowski** [comment]
Rebased, bumped to 1.41.1

**Comment by mjankowski** [comment]
I was about to start dealing with some of the violations from https://github.com/mastodon/mastodon/pull/32508 -- but I'd much rather progress this one if possible.

I'd love to get feedback on this direction. In my opinion it's a solid removal of a bunch of stuff I don't think we should actually care about, and I think it will provide a simpler upgrade/migration/consistency path in the future.

Also if we DONT want to go this direction, there are a bunch of linked/related things I can try to push forward on their own where we currently have misconfigurations.

**Comment by mjankowski** [comment]
This PR is almost 2 years old and has been closed for some time, but FYI I rebased/updated the spirit of this, described in https://github.com/mastodon/mastodon/pull/37988#issuecomment-3967212426

I can revive this if there's interest.

**Pull request closed without merging**